rust/src/storage/file/mod.rs

```diff
@@ -87,33 +84,70 @@ impl StorageBackend for FileStorageBackend {
     async fn list_objs<'a>(
         &'a self,
         path: &'a str,
-    ) -> Result<
-        Pin<Box<dyn Stream<Item = Result<ObjectMeta, StorageError>> + Send + 'a>>,
-        StorageError,
-    > {
-        let readdir = ReadDirStream::new(fs::read_dir(path).await?);
-
-        Ok(Box::pin(readdir.err_into().and_then(|entry| async move {
-            let path = String::from(
-                entry
-                    .path()
-                    .to_str()
-                    .ok_or_else(|| StorageError::Generic("invalid path".to_string()))?,
-            );
-            match entry.metadata().await {
-                Ok(meta) => {
-                    Ok(ObjectMeta {
-                        path,
-                        modified: meta.modified()?.into(),
-                        size: Some(meta.len().try_into().map_err(|_| {
-                            StorageError::Generic("cannot convert to i64".to_string())
-                        })?),
-                    })
+    ) -> Result<BoxStream<'a, Result<ObjectMeta, StorageError>>, StorageError> {
+        let walkdir = WalkDir::new(path)
+            // Don't include the root directory itself
+            .min_depth(1);
+
+        let s =
+            walkdir.into_iter().flat_map(move |result_dir_entry| {
```

Comment:
Same thing over here: I think `filter_map` makes more sense for this case:

```suggestion
            walkdir.into_iter().filter_map(move |result_dir_entry| {
```

Reply:
done.